src/core/stdcpp/vector.d

```diff
@@ -114,6 +114,17 @@ extern(D):
     ///
     void append(T[] array)                                                  { insert(length, array); }
 
+    /// Performs elementwise equality check.
+    bool opEquals(this This, That)(auto ref That rhs)
+    if (is(immutable That == immutable vector))                             { return as_array == rhs.as_array; }
+
+    /// Performs lexicographical comparison.
+    int opCmp(this This, That)(auto ref That rhs)
+    if (is(immutable That == immutable vector))                             { return __cmp(as_array, rhs.as_array); }
+
+    /// Hash to allow `vector`s to be used as keys for built-in associative arrays.
+    /// **The result will generally not be the same as C++ `std::hash<std::vector<T>>`.**
+    size_t toHash() const                                                   { return .hashOf(as_array); }
```

Comment:
`toHash` on the other hand needs to be a non-template const function or else it will be ignored by the compiler for the purpose of determining if a particular struct has a non-default hash code.

Reply:
Please leave an obvious TODO that we should use std::hash in the future when it's written?